src/utils/Validator.js

```diff
@@ -0,0 +1,35 @@
+/*
+오류
+- 받은값이 음수일때
+- 인식할수없는 값이 들어왔을때 (구분자를 추가했는데도 인식할수없는 구분자가 있는 등)
+- 커스텀 구분자가 있긴한데 안에 있는게 문자가 아니라 문자들인경우
+
+*/
+
+class Validator {
+  static validateNumber(number) {
+    if (Number.isNaN(number) || number < 0) {
+      throw new Error('[ERROR] 유효하지 않은 입력값입니다.');
+    }
+    return number;
+  }
+
+  // 구분자 없앤 string 배열
+  static checkArray(numberArray) {
+    let isNegative = false;
+    let isNotNumber = false;
+
+    if (numberArray.length === 1 || numberArray[0] === '') {
+      return;
+    }
+
+    for (let num = 0; num < numberArray.length; num += 1) {
+      if (parseInt(numberArray[num], 10) < 0) isNegative = true;
+      if (numberArray[num] < '0' || numberArray[num] > '9') isNotNumber = true;
+    }
+    if (isNegative) throw new Error('[ERROR] 입력 값 중 음수가 존재합니다.');
+    if (isNotNumber) throw new Error('[ERROR] 유효하지 않은 입력 값 입니다.');
+  }
```

Comment:
```suggestion
    for (const num of numberArray) {
      if (parseInt(num, 10) < 0) {
        throw new Error(...);
      if (num < '0' || num > '9') {
      throw new Error(...); 
    }
  }
```

이런식으로 고치시면 인덱스로 보지 않아도 되고, 조건상 배열을 다 순회하지 않아도 괜찮을것 같아요!

Reply:
헉 코드가 훨씬 직관적이고 보기 편해지네요....! for of를 평소에 사용하지 않아서 생각하지 못했었는데 이렇게 지적해주셔서 되돌아보게됩니다. 코드 작성해주시면서까지 피드백해주셔서 감사해요 참고하겠습니다!